File: fileprovider-camera/MyApplication/app/src/main/java/com/example/se/myapplication/MainActivity.java
package com.example.se.myapplication;

import android.content.Intent;
import android.graphics.Bitmap;
import android.net.Uri;
import android.os.Bundle;
import android.provider.MediaStore;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private static final int REQUEST_IMAGE_CAPTURE = 1;
    private TextView imageUriTextView;
    private ImageView imageView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        imageUriTextView = (TextView) findViewById(R.id.imageUriTextView);
        imageView = (ImageView) findViewById(R.id.imageView);

        findViewById(R.id.cameraButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
            }
        });

        findViewById(R.id.galleryButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
                if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
                    startActivityForResult(takePictureIntent, REQUEST_IMAGE_CAPTURE);
                }
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if (requestCode == REQUEST_IMAGE_CAPTURE && resultCode == RESULT_OK) {
            Bundle extras = data.getExtras();

            Uri uri = data.getData();
            imageUriTextView.setText(String.valueOf(uri));

            Bitmap imageBitmap = (Bitmap) extras.get("data");
            imageView.setImageBitmap(imageBitmap);
        }
    }
}

File: fileprovider-camera/appcrop/src/main/java/siarhei/luskanau/example/camera/appcrop/MainActivity.kt
package siarhei.luskanau.example.camera.appcrop

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.core.graphics.drawable.RoundedBitmapDrawableFactory
import com.miguelbcr.ui.rx_paparazzo2.RxPaparazzo
import com.yalantis.ucrop.UCrop
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.rxkotlin.subscribeBy
import io.reactivex.schedulers.Schedulers
import java.util.Locale
import siarhei.luskanau.example.camera.appcrop.databinding.ActivityMainBinding
import timber.log.Timber

private const val MAX_BITMAP_SIZE = 1024

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        ActivityMainBinding.inflate(LayoutInflater.from(this))
            .also {
                binding = it
                setContentView(binding.root)
            }

        findViewById<View>(R.id.cameraButton).setOnClickListener {
            RxPaparazzo.single(this)
                .useInternalStorage()
                .crop(getCropOptions())
                .usingCamera()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeBy(
                    onNext = { showImageUri(Uri.fromFile(it.data().file)) },
                    onError = { Timber.e(it) }
                )
        }

        findViewById<View>(R.id.galleryButton).setOnClickListener {
            RxPaparazzo.single(this)
                .useInternalStorage()
                .crop(getCropOptions())
                .usingGallery()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeBy(
                    onNext = { showImageUri(Uri.fromFile(it.data().file)) },
                    onError = { Timber.e(it) }
                )
        }

        showImageUri(null)
    }

    private fun getCropOptions() = UCrop.Options().apply {
        setToolbarColor(ContextCompat.getColor(applicationContext, android.R.color.white))
        setStatusBarColor(ContextCompat.getColor(applicationContext, android.R.color.white))
        setToolbarWidgetColor(ContextCompat.getColor(applicationContext, android.R.color.black))
        setToolbarTitle("")
        withAspectRatio(1f, 1f)
        setShowCropGrid(false)
        setCompressionFormat(Bitmap.CompressFormat.PNG)
        setMaxBitmapSize(MAX_BITMAP_SIZE)
        setHideBottomControls(true)
    }

    private fun showImageUri(uri: Uri?) {
        binding.imageUriTextView.text = String.format(Locale.ENGLISH, "Uri: %s", uri.toString())
        uri?.let {
            val bitmap = BitmapFactory.decodeStream(contentResolver.openInputStream(uri))
            val roundedBitmapDrawable = RoundedBitmapDrawableFactory.create(resources, bitmap)
                .apply { isCircular = true }
            binding.imageView.setImageDrawable(roundedBitmapDrawable)
            binding.circularImageView.setImageURI(uri)
        } ?: run {
            binding.imageView.setImageResource(R.drawable.ic_android_24dp)
            binding.circularImageView.setImageURI(uri)
        }
    }
}
